node/src/crypto/asymmetric_key.rs

```diff
@@ -739,23 +739,24 @@ impl ToBytes for PublicKey {
         match self {
             PublicKey::Ed25519(public_key) => {
                 buffer.insert(0, ED25519_TAG);
-                buffer.extend(public_key.as_ref().to_vec().into_bytes()?);
+                buffer.extend(bytesrepr::serialize_bytes(public_key.as_ref())?);
```

Comment:
If you make the changes suggested for the new `serialize_` and `deserialize_` functions in bytesrepr, this will need changed.

I'm not sure why I decided to serialize the `PublicKey` type as a `Vec<u8>` - it's a bad decision.  We already have const member variables `ED25519_LENGTH` and `SECP256K1_LENGTH`, meaning we should construct an array here and serialize that.  That should probably be done regardless of whether the changes I suggested for bytesrepr are adopted or not.

Same comment goes for `Signature` in this file (although the Ed25519 case is already an array, but it'll need changed anyway as it's just calling `to_bytes` and that's inefficient, and will fail if you add the `debug_assert`s I suggested)